traffic_ops/traffic_ops_golang/crconfig/edgelocations.go

```diff
@@ -57,6 +57,36 @@ and (st.name = 'REPORTED' or st.name = 'ONLINE' or st.name = 'ADMIN_DOWN')
 		if ttype == RouterTypeName {
 			routerLocs[cachegroup] = latlon
 		} else {
+			primaryCacheId := ""
```

Comment:
rather than re-querying the cachegroup's ID here, you should just update the query on L35 to also select `cg.id` and scan that into a variable.

In fact, you might even be able to eliminate the query on L65 altogether by modifying the query on L35 to include a subquery. Something like:
```
select cg.name, ..., ARRAY(select c.name from cachegroup_fallbacks cgf join cachegroup c on c.id = cgf.backup_cg where cgf.primary_cg = cg.id) ... from ...
```
I haven't tried that query out, but I think it's worth a shot rather than running an extra query per cachegroup. If it's possible, you should be able to scan that result into a slice of strings and just iterate over it.